packages/smart-wallet/src/smartWallet.js

```diff
@@ -990,21 +991,23 @@ export const prepareSmartWallet = (baggage, shared) => {
             // await so that any errors are caught and handled below
             await watchOfferOutcomes(watcher, seatRef);
           } catch (err) {
-            facets.helper.logWalletError('OFFER ERROR:', err);
+            // This block only runs if the block above fails during one vat incarnation.
+            facets.helper.logWalletError('IMMEDIATE OFFER ERROR:', err);
 
-            // Notify the user
+            // Update status to observers
             if (err.upgradeMessage === 'vat upgraded') {
               // The offer watchers will reconnect. Don't reclaim or exit
               return;
             } else if (watcher) {
-              watcher.helper.updateStatus({ error: err.toString() });
+              // The watcher's onRejected will updateStatus()
             } else {
               facets.helper.updateStatus({
                 error: err.toString(),
                 ...offerSpec,
               });
             }
 
+            // Backstop recovery, in case something very basic fails.
             if (offerSpec?.proposal?.give) {
               facets.payments
                 .tryReclaimingWithdrawnPayments(offerSpec.id)
```

Comment:
So we assume this case only matters for immediate rejections from the seat?

I suppose that `tryReclaimingWithdrawnPayments` is not very idempotent, nor parallel safe.

Reply:
"parallel"?

Noting that at https://github.com/Agoric/agoric-sdk/pull/8488 I see 

> along the way `payments.tryReclaimingWithdrawnPayments()` went away. I'm pretty sure it was redundant with ERTP recoverySets.